Cosmostation/Base/AppDelegate.swift

```diff
@@ -84,15 +84,12 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
     }
 
     func applicationDidEnterBackground(_ application: UIApplication) {
-        if(application.topViewController!.isKind(of: IntroViewController.self) ||
-            application.topViewController!.isKind(of: PasswordViewController.self)) {
-            
-        } else {
-            if (BaseData.instance.getUsingAppLock() && BaseData.instance.hasPassword()) {
+        if let topViewController = application.topViewController {
+            if !topViewController.isKind(of: PasswordViewController.self) && BaseData.instance.isRequiredUnlock() {
```

Comment:
I have checked that the only condition needed to show the PasswordViewController is that the app has a lock and a password was set. 
You can check in the old code that most of the times `if(application.topViewController!.isKind(of: IntroViewController.self)` is not met as topViewController was an UINavigationController(fixed later)